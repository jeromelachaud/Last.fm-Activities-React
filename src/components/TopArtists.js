import React from 'react';
import styled from 'styled-components';
import TopArtist from './TopArtist.js';
import _map from 'lodash/map';
import {
  fetchTopArtists
} from '../actions/action-creators';

const TopArtistsWrapper = styled.section`
  width: 30rem;
  margin: 0 auto
`;
let TopArtists = React.createClass ({
  propTypes: {
    topArtists: React.PropTypes.array
  },

  componentWillMount() {
    fetchTopArtists();
  },

  render() {
    const {
      topArtists
    } = this.props;

    let TopArtistElement;
    if (topArtists) {
      TopArtistElement = _map(topArtists, (topArtist, i) => (
        <TopArtist
          key={i}
          topArtist={topArtist} />
      ));
    }

    return (
    <TopArtistsWrapper>
      {TopArtistElement}
    </TopArtistsWrapper>
    );
  }
});

export default TopArtists;
