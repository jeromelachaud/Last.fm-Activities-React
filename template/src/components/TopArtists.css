/* TopArtists */
.TopArtists{
  width: 30rem;
  margin: 0 auto
}
