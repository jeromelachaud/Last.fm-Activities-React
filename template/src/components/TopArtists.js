import React from 'react';
import './TopArtists.css';
import TopArtist from './TopArtist.js';
import _map from 'lodash/map';

let TopArtists = React.createClass ({

  render() {
    const {
      topArtists
    } = this.props;

    // let TopArtistElement;
    // if (topArtists) {
    //   TopArtistElement = _map(topArtists, (recentTrack, i) => (
    //     <TopArtist
    //       key={i}
    //       recentTrack={recentTrack} />
    //   ));
    // }

    return (
    <section
      className="TopArtists">
      {/* {TopArtistElement} */}
      TopArtists
    </section>
    );
  }
});

export default TopArtists;
