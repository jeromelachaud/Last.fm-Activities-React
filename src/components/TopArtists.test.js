import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import TopArtists from './TopArtists';

describe('<TopArtists> component', () => {
  const wrapper = shallow(<TopArtists/>);

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
