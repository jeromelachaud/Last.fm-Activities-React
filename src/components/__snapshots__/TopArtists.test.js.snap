exports[`<TopArtists> component should render as expected 1`] = `<styled.section />`;
